tools/RcclReplayer/rcclReplayer.cpp

```diff
@@ -2,757 +2,627 @@
-    }
-    NCCL_CALL(ncclGroupEnd());
-  }
-  printf("Rank %d Done setting up communicators\n", mpiRank);
-
-  int numSkippedCalls = 0;
-  int numInvalid = 0;
-  double runTime;
-  std::ofstream datafile;
-  datafile.open("replayer_data.csv");
-  if (!datafile.is_open()) {
-    printf("[ERROR] Unable to open file replayer_data.csv\n");
-    exit(-1);
-  }
-  datafile << "callNumber, functionName, inPlace, count(numElements), datatype, op, root, time(msec), groupCallBusBandwidth(GB/s)\n";
-  auto start = std::chrono::high_resolution_clock::now();
-  for (size_t i = 0; i < collCalls.groupCalls.size(); i++) {
-    MPI_Barrier(MPI_COMM_WORLD);
-    if (collCalls.groupCalls[i].isValid) {
-      if (mpiRank == 0)
-      {
-        printf("Running Collective Call %lu of %lu\n", i+1, collCalls.groupCalls.size());
-        PrintGroupCall(collCalls.groupCalls[i]);
-      }
-      double runTime = ReplayRccl(collCalls, i, numInvalid);
-      if (mpiRank == 0) {
-        dataToCsv(collCalls.groupCalls[i], datafile, runTime);
-      }
-    } else {
-      if (mpiRank == 0) {
-        printf("[ERROR] in group call: (skipping...)\n");
-        for (auto const& rd : collCalls.groupCalls[i].rankData) {
-          printf("  - Rank %02d: comm %d in line %d\n", rd.first, rd.second.commIdx, rd.second.lineNum);
-          for (int task = 0; task < rd.second.tasks.size(); task++) {
-            TaskInfo ti = rd.second.tasks[task];
-            printf("  - Task %02d: %32s inPlace=%d count=%lu datatype=%d op=%d root=%d\n",
-                   task, ncclFuncNames[ti.funcType], ti.inPlace, ti.count, ti.datatype, ti.op, ti.root);
-          }
-        }
-      }
-      numSkippedCalls++;
-    }
-  }
-  auto end = std::chrono::high_resolution_clock::now();
-  std::chrono::duration<double> duration = end - start;
-  datafile.close();
+#include <dirent.h>
+#include <stdio.h>
 
-  // Destroy all communicators
-  for (int commIdx = 0; commIdx < collCalls.numCommsPerRank; commIdx++) {
-    for (int i = 0; i < numGpusPerMpiRank; i++) {
-      NCCL_CALL(ncclCommDestroy(collCalls.localRankComms[i][commIdx]));
-      HIP_CALL(hipStreamDestroy(collCalls.localRankStreams[i][commIdx]));
-    }
-  }
-
-  if (mpiRank == 0) printf("Executed group calls: %zu\n", collCalls.groupCalls.size() - numSkippedCalls);
-  if (mpiRank == 0) printf("Skipped group calls: %d\n", numSkippedCalls);
-
-  // Data validation failures during group calls
-  if (mpiRank == 0) printf("Failed group calls: %d\n", numInvalid); 
+using namespace rccl;
 
-  // Time it takes to execute all the group calls
-  if (mpiRank == 0) printf("Execution Time: %f seconds\n", duration.count());
-  printf("MPI Rank %d Success\n", mpiRank);
+static int json_format = 0; // binary by default
 
-  MPI_Finalize();
-  return 0;
-}
+// move to inside class or kept as static var
+static constexpr size_t rcclCallSize = sizeof(rcclApiCall);
+static char line[rcclCallSize]; // size of collectivecall struct
```

Comment:
Why isn't this just a local variable?

shouldn't you need at least +1 to the size to handle C strings?